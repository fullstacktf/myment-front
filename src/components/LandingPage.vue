<template>
  <div class="landing">
    <div class="leftZone">
      <log-menu></log-menu>
    </div>
    <div class="rightZone">
      <anim-logo></anim-logo>
      <to-short name="Let's beggin" customWidth="250px" customHeight="100px"></to-short>
    </div>
  </div>
</template>

<script>
import LoginMenu from './LoginMenu';
import AnimLogo from './AnimLogo';
import ToShort from './Button';

export default {
  name: 'LandingPage',
  components: {
    'log-menu': LoginMenu,
    'anim-logo': AnimLogo,
    'to-short': ToShort,
  },
};
</script>

<style>
.landing {
  display: grid;
  grid-template-columns: 20% 80%;
  width: 100vw;
  height: 100vh;
}
.rightZone {
  display: grid;
  justify-items: center;
  align-items: center;
  grid-template-rows: 80% 20%;
}
</style>
