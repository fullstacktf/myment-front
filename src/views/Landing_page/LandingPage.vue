<template>
  <div class="landing">
    <div class="leftZone">
      <sign-menu></sign-menu>
    </div>
    <div class="rightZone">
      <anim-logo></anim-logo>
      <router-link to="/short">
        <to-short name="Let's beggin" customWidth="700px" customHeight="60px"></to-short>
      </router-link>
    </div>
  </div>
</template>

<script>
import SignUpMenu from '../../components/SignUpMenu';
import AnimLogo from '../../components/AnimLogo';
import ToShort from '../../components/Button';

export default {
  name: 'LandingPage',
  components: {
    'sign-menu': SignUpMenu,
    'anim-logo': AnimLogo,
    'to-short': ToShort,
  },
};
</script>

<style>
.landing {
  display: grid;
  grid-template-columns: 20% 80%;
  width: 100vw;
  height: 100vh;
  background-color: #66838c;
}
.rightZone {
  display: grid;
  justify-items: center;
  align-items: center;
  grid-template-rows: 80% 20%;
}
</style>
