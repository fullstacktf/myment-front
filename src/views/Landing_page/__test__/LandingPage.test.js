import LandingPage from '../LandingPage.vue';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import Buefy from 'buefy';
import VueRouter from 'vue-router';
import VueMouseParallax from 'vue-mouse-parallax';

describe('LandingPage', () => {
  let localVue;
  beforeEach(() => {
    const Vue = createLocalVue();
    Vue.use(Buefy);
    Vue.use(VueRouter);
    Vue.use(VueMouseParallax);
    localVue = Vue;
  });
  test('render component', () => {
    const wrapper = shallowMount(LandingPage, {
      localVue,
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('Check props', () => {});

  test('check Contents props', () => {});
});
