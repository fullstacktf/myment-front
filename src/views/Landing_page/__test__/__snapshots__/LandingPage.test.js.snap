// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage render component 1`] = `"<div class=\\"columns is-centered landingParent\\"><div class=\\"mustBeOnTop\\"><sign-menu-stub></sign-menu-stub></div> <main id=\\"page-wrap\\"><parallax-container-stub class=\\"childWrapper\\"><parallax-element-stub type=\\"depth\\" parallaxstrength=\\"10\\"><div class=\\"columns\\"><div class=\\"column is-1\\"></div> <div class=\\"column is-4\\"><div class=\\"columnContent\\"><parallax-container-stub><parallax-element-stub type=\\"depth\\" parallaxstrength=\\"10\\"><div class=\\"imageWrappergif\\"><img src=\\"../../assets/giphy.gif\\"></div></parallax-element-stub></parallax-container-stub> <parallax-container-stub><parallax-element-stub type=\\"depth\\" parallaxstrength=\\"10\\"><h1 class=\\"title is-1 is-white myment\\">M Y M E N T</h1></parallax-element-stub></parallax-container-stub></div></div> <div class=\\"column is-8\\"><div class=\\"sloganWrapper\\"><parallax-container-stub><parallax-element-stub type=\\"depth\\" parallaxstrength=\\"20\\"><h1 class=\\"title is-3 is-white\\">Start using the best itinerary generator</h1></parallax-element-stub></parallax-container-stub> <parallax-container-stub><parallax-element-stub type=\\"depth\\" parallaxstrength=\\"15\\"><h2 class=\\"subtitle is-3 is-white\\">Discover that city where you have never been, now easier with myment, just by adding your preferences. Start now quicker than ever and without loggin.</h2></parallax-element-stub></parallax-container-stub> <parallax-container-stub><parallax-element-stub type=\\"depth\\" parallaxstrength=\\"15\\"><router-link-stub to=\\"/short\\" tag=\\"a\\" event=\\"click\\"><button class=\\"button main is-large is-fullwidth is-warning is-rounded is-focused\\">LET'S GO FOR IT</button></router-link-stub></parallax-element-stub></parallax-container-stub></div></div></div></parallax-element-stub></parallax-container-stub></main></div>"`;
